### ISSUE TITLE
Spectral discotinuities when processing sub-bands in parallel

### ISSUE THREAD

**Issue by paoloserra** [caracal-pipeline/caracal#1245]
When reducing the data in sub-bands we often worry about spectral discontinuities at the boundary between sub-bands.

Here I report on the case of a CARACal data reduction where we split the data in 10 4-MHz-wide sub-bands after self-calibration. All processing steps after self-calibration all the way to line imaging are done independently on each sub-band.

In this case, the only potential source of spectral discontinuity is continuum subtraction with 1st-order UVLIN.

Individual line spectra show no discontinuities. However, these may be hard to see on top of the noise. For this reason I stacked 1,000 spectra extracted at random X,Y positions. The stacked spectrum is attached. The vertical red lines represent sub-band boundaries.

I see no discontinuities in this case.

![discont](https://user-images.githubusercontent.com/6591265/93893019-747cb500-fced-11ea-9729-3b63efc60d01.png)

I'd be interested to know if there are additional/better ways of investigating this.

Tagging a few people who might be interested or commented on this in the past. @edeblok @gigjozsa @SpheMakh

**Comment by KshitijT** [comment]
Would it be possible to overlay a "mean/DC level" per subband? Not that there is any reason to expect any funny business since calibration is already done..

**Comment by paoloserra** [comment]
Good idea -- calibration is done indeed (and on the full band), but that mean level will be modified by UVLIN per sub-band.

Of course, no sub-band will have exactly zero mean level, and there will be differences between sub-bands. But I wonder whether you already have an idea of what would be an acceptable difference.

**Comment by paoloserra** [comment]
@KshitijT here it is. The median within each sub-band is shown by the blue line. If you see no blue line it means it's under the zero-level red line. As I guessed yesterday, there are small differences of course. They do seem negligible to me, but I'm not sure how to make the latter statement more quantitative. For sure the difference between adjacent sub-bands is a lot less than the noise in the stacked spectrum.

![discont_dc](https://user-images.githubusercontent.com/6591265/93998492-e4924600-fd94-11ea-88b8-0c6ea7cff969.png)

**Comment by KshitijT** [comment]
>  For sure the difference between adjacent sub-bands is a lot less than the noise in the stacked spectrum.

I guess that can be the metric to go by? The difference ARE really tiny..

**Comment by paoloserra** [comment]
Right, the differences are negligible if they are less than the rms, but probably the correct comparison should be with rms/sqrt(nrchans_in_subband)

**Comment by KshitijT** [comment]
An idle thought - it would be interesting to see if the piece-wise, 1st Order UVLIN continuum subtraction does better for a source with spectral curvature and/or a source further away from the phase centre (thinking of the primary-beam here). Naively, it should do better for a piece-wise subtraction, but would it then give DC offsets?
